### PR TITLE
feat(platform): Provide `repoFingerprint` field on repository init

### PR DESCRIPTION
**Pull request by zharinov**
## Context

- Ref: #17247

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [x] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/platform/types.ts [hunk 1]
```diff
@@ -28,6 +28,7 @@ export interface PlatformResult {
 export interface RepoResult {
   defaultBranch: string;
   isFork: boolean;
+  repoCacheId: string;
```
Comment (rarkins) [inline lib/modules/platform/types.ts:31]:
```suggestion
  repoId: string;
```

The platform layer isn't concerned with whether this ID is for cache or other reasons

**Review by rarkins — CHANGES_REQUESTED** [review]

File: lib/modules/platform/types.ts (hunk 1 above)
Comment (zharinov) [reply lib/modules/platform/types.ts:31]: Now I think it's better to call it `fingerprint` or `repoFingerprint`, as we probably want to distinguish not only `id` but also `endpoint`

Comment (viceice) [reply lib/modules/platform/types.ts:31]: sound good

File: lib/modules/platform/azure/__snapshots__/index.spec.ts.snap [hunk 2]
```diff
@@ -280,6 +280,7 @@ exports[`modules/platform/azure/index initRepo should initialise the config for
 {
   "defaultBranch": "defBr",
   "isFork": false,
+  "repoFingerprint": "02574de485149547c1a071aa7921da3d0afadcd6162f3bd49ba3ced29be589f8b9bac689fd0badb212bd21c3f48bd8566beaf31cdca2b083bd855808a9c129e2",
```
Comment (viceice) [inline lib/modules/platform/azure/__snapshots__/index.spec.ts.snap:283]: we should get rid of those snapshots in separate PR

File: lib/modules/platform/util.ts [hunk 3]
```diff
@@ -35,3 +36,12 @@ export function detectPlatform(url: string): 'gitlab' | 'github' | null {
 
   return null;
 }
+
+export function repoFingerprint(
```
[the remark this answers is not in the document]
Comment (zharinov) [reply lib/modules/platform/util.ts:40]: I'd argue for current more specific signature which makes it more clear what inputs we're expecting for repo fingerprinting.

File: lib/modules/platform/azure/__snapshots__/index.spec.ts.snap (hunk 2 above)
Comment (zharinov) [reply lib/modules/platform/azure/__snapshots__/index.spec.ts.snap:283]: Okay

Comment (zharinov) [reply lib/modules/platform/azure/__snapshots__/index.spec.ts.snap:283]: But let's do this after this one is merged, since it'll take a while

**Review by rarkins — APPROVED** [review]

**Review by viceice — APPROVED** [review]

**Pull request merged**